docker/update_nvidia_keys.sh

```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
+
+{
+    apt-get update -y && apt-get install -yq gnupg wget &&\
+    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+
+} || {
+
+    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
+    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+
+}
```

Comment:
why the somewhat complicated if? Does `apt install gnupg wget` fail on the cuda image?

If it does, I think the following is a lot shorter/clearer
```suggestion
    apt update -y
    apt install -yq gnupg wget || true
    # https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/
    apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
```

Reply:
I guess that wasn't what I was asking. My question was if the `apt install -yq gnupg` ever fails. If not, why do we need the `|| true`?
Anyway, I don't mind having it.

Reply:
ok. I think I understand. On the GPU image, anything with `apt` fails before adding the keys. sorry.